### ISSUE TITLE
White-box Cartoon 训练过程的问题

### ISSUE THREAD

**Issue by AlainConnor** [zhen8838/AnimeStylized#4]
您好，
非常感谢您能开源代码。
在使用过程中遇到了一点小问题，想与您交流一下。

我测试的代码是《White-box Cartoon》这篇论文的复现代码。

在train的时候，会报一个错误。
这个错误的位置是whiteboxgan_pretrain.py的30-31行，看了一下dataset的构造和后序训练的输入，在30和31行之间，是否应当增加
```
input_photo = torch.cat(input_photo, dim=0)
```

否则input_photo是一个元组有两个元素，没有进行合并。

**Comment by zhen8838** [comment]
恩恩，之前代码代码迭代的太快，我没再次运行过pretrain。validation_step里面的代码我刚刚已经修改了，原本我是只检查景色图像，现在是检查景色和人脸图像。

**Comment by AlainConnor** [comment]
可以了解一下您的硬件环境吗？
显卡、cpu、和内存，以及训练风格迁移模型需要大约多少小时？
我最近在研究这一块，谢谢老哥

**Comment by zhen8838** [comment]
我用的是v100，内存32g，animegan和whiteboxgan一般晚上10点训练到早上9点就可以得到比较好的模型了。我给的预训练模型都是训练一晚上的结果。animegan的学习率非常低，如果要快速见效可以提高一下。whiteboxgan需要cpu去做superpixel就比较慢，都在等待cpu处理，实际训练时间其实比较短，优化速度的话，可以考虑先把数据集都superpixel预处理一下。

**Comment by AlainConnor** [comment]
好的，多谢！

**Comment by zhen8838** [comment]
不用谢，我最近在研究如何用轻量级的风格迁移进行人脸动漫化，如果有兴趣可以一起研究。当然也欢迎贡献代码

**Comment by AlainConnor** [comment]
可以呀，我先把常见的卡通化论文看一遍。
您那边是有讨论小组吗？

**Comment by zhen8838** [comment]
那到没有，我目前自己研究，现在做了些实验，也在看论文中

**Comment by AlainConnor** [comment]
噢噢，老哥有空能发一下您看过的相关论文列表嘛，不胜感激啊！

**Comment by zhen8838** [comment]
- U-GAT-IT: UNSUPERVISED GENERATIVE ATTENTIONAL NETWORKS WITH ADAPTIVE L AYER INSTANCE NORMALIZATION FOR IMAGE-TO-IMAGE TRANSLATION
- CartoonGAN: Generative Adversarial Networks for Photo Cartoonization
- CycleGAN Face-off
- Face-to-Parameter Translation for Game Character Auto-Creation
- Fast Face-swap Using Convolutional Neural Networks
- Learning from Multi-domain Artistic Images for Arbitrary Style Transfer
- Parameter-Free Style Projection for Arbitrary Style Transfer
- Encoding in Style: a StyleGAN Encoder for Image-to-Image Translation
- StyleRemix: An Interpretable Representation for Neural Image Style Transfer
- Unpaired Photo-to-manga Translation Based on The Methodology of Manga Drawing
- EVERYONE IS A CARTOONIST: SELFIE CARTOONIZATION WITH ATTENTIVE ADVERSARIAL NETWORKS
- Visual Attribute Transfer through Deep Image Analogy 

- DESIGN-GAN: CROSS-CATEGORY FASHION TRANSLATION DRIVEN BY LANDMARK ATTENTION

**Comment by AlainConnor** [comment]
感谢老哥，读完与你交流！

**Comment by huangfuyang** [comment]
> 我用的是v100，内存32g，animegan和whiteboxgan一般晚上10点训练到早上9点就可以得到比较好的模型了。我给的预训练模型都是训练一晚上的结果。animegan的学习率非常低，如果要快速见效可以提高一下。whiteboxgan需要cpu去做superpixel就比较慢，都在等待cpu处理，实际训练时间其实比较短，优化速度的话，可以考虑先把数据集都superpixel预处理一下。

数据集superpixel预处理有什么用处么？structure loss的算的不是生成出来的图片的superpixel么？ F(G(Ip))。算原始图的superpixel也没啥用吧

**Comment by zhen8838** [comment]
嗯，是没用，上次疏忽了，写错了。


发自我的iPhone

------------------ 原始邮件 ------------------
发件人: huangfuyang ***@***.***&gt;
发送时间: 2021年4月8日 16:45
收件人: zhen8838/AnimeStylized ***@***.***&gt;
抄送: 郑启航 ***@***.***&gt;, Comment ***@***.***&gt;
主题: 回复：[zhen8838/AnimeStylized] White-box Cartoon 训练过程的问题 (#4)





  
我用的是v100，内存32g，animegan和whiteboxgan一般晚上10点训练到早上9点就可以得到比较好的模型了。我给的预训练模型都是训练一晚上的结果。animegan的学习率非常低，如果要快速见效可以提高一下。whiteboxgan需要cpu去做superpixel就比较慢，都在等待cpu处理，实际训练时间其实比较短，优化速度的话，可以考虑先把数据集都superpixel预处理一下。
  
数据集superpixel预处理有什么用处么？structure loss的算的不是生成出来的图片的superpixel么？ F(G(Ip))。算原始图的superpixel也没啥用吧
